### PR TITLE
Add "install" target

### PR DESCRIPTION
**Pull request by androm3da**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,19 +42,30 @@ test_processors: test_processors.S hexagon_vm.h Makefile
...
-.PHONY: dbg
+.PHONY: dbg install
 dbg: FORCE
 	lldb -o 'file ./minivm' -o 'target modules add ./vmlinux' -o 'target modules load -s 0 --file ./vmlinux' -o 'gdb-remote localhost:1234' ${LLDB_OPTS}
 
 minivm.bin: minivm
 	${OBJCOPY} -O binary $< $@
 
+install: minivm minivm.bin $(TEST_CASES)
+	 install -D $^ $(DESTDIR)$(bindir)
```
Comment (quic-mathbern) [inline Makefile]: Should we default DESTDIR to something like "install" if it is not defined? Or even error out?

Also, bindir doesn't seem to be defined?

Comment (androm3da) [reply Makefile]: If `DESTDIR` is substituted as the empty string then the implicit behavior would be that the user wants them installed to the root.  It's similar-but-not-quite-the-same as the `prefix` / `--prefix`.

Hmm, I guess, now that I'm thinking about it - `prefix` is the Right Thing to do here, so I'll switch to that.

See https://www.chiark.greenend.org.uk/doc/make-doc/make.html/Makefile-Conventions.html#Directory-Variables and https://www.chiark.greenend.org.uk/doc/make-doc/make.html/Makefile-Conventions.html#DESTDIR

Comment (androm3da) [reply Makefile]: > Also, bindir doesn't seem to be defined?

I'll fix this

File: Makefile [hunk 2]
```diff
@@ -51,13 +53,17 @@ run-%: tests_bin/% minivm
...
 dbg: FORCE
 	lldb -o 'file ./minivm' -o 'target modules add ./vmlinux' -o 'target modules load -s 0 --file ./vmlinux' -o 'gdb-remote localhost:1234' ${LLDB_OPTS}
 
 minivm.bin: minivm
 	${OBJCOPY} -O binary $< $@
 
+install: minivm minivm.bin $(TESTS_BIN)
+	 mkdir -p $(prefix)$(bindir)/
+	 install -D $^ $(prefix)$(bindir)/
+
```
Comment (quic-mathbern) [inline Makefile]: https://www.chiark.greenend.org.uk/doc/make-doc/make.html/Makefile-Conventions.html#Directory-Variables says:

> The default value of prefix should be /usr/local

and

> The default value of exec_prefix should be $(prefix)

Shouldn't we do something like this at the top?

```
prefix?=/usr/local
exec_prefix?=$(prefix)
bindir?=$(exec_prefix)/bin
```

And then, here at install:

```
mkdir -p $(bindir)
install -D $^ $(bindir)
```

**Review by quic-mathbern — APPROVED** [review]

**Pull request merged**